crates/uv/src/commands/pip/operations.rs

```diff
@@ -210,6 +212,56 @@ pub(crate) async fn resolve<InstalledPackages: InstalledPackagesProvider>(
             );
         }
 
+        for (pyproject_path, groups) in groups {
+            let empty = PathBuf::new();
+            let pyproject_parent = pyproject_path.parent().unwrap_or(&empty);
+            // TODO(Gankra): this surely isn't right but VirtualProject::discover asserts on relative paths... hrm.
+            let project_dir = current_dir().unwrap().join(pyproject_parent);
+            let metadata = SourcedDependencyGroups::from_virtual_project(
```

Comment:
TODO....

Reply:
I think the assert is telling me I'm *still* using the wrong API though:

> This method requires an absolute path and panics otherwise, i.e. this method only supports discovering the main workspace.

I think this is saying it won't properly detect a package under a workspace?